src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -144,6 +146,11 @@ var ReactDOMInput = {
       listeners: null,
       onChange: _handleChange.bind(inst),
     };
+
+    if (__DEV__) {
+      var controlledValue = props.type === 'checkbox' ? props.checked : props.value;
```

Comment:
checked is also valid for radio.  We probably shouldn't be branching based on props.type.  I think we can just do `controlled = props.checked !== undefined || props.value != undefined`